### ISSUE TITLE
fails to start when it encounters a broken symlink

### ISSUE THREAD

**Issue by aaronjensen** [castwide/solargraph#308]
I've worked with a few gems that have inadvertently included symlinks in the gem to files that do not exist within the gem. When solargraph encounters these, it appears to stop its initialization process.

Here's an example:

```
[WARN] Error processing request: [Errno::ENOENT] No such file or directory @ rb_sysopen - ~/Source/project/es-sync-sandbox/count-component/gems/ruby/2.7.0/gems/evt-consumer-2.2.0.1/lib/consumer/postgres.rb
[WARN] ~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/source.rb:491:in `initialize'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/source.rb:491:in `open'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/source.rb:491:in `load'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/workspace.rb:148:in `block in load_sources'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/workspace.rb:147:in `each'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/workspace.rb:147:in `load_sources'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/workspace.rb:27:in `initialize'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/library.rb:363:in `new'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/library.rb:363:in `load'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/host.rb:269:in `prepare'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/host.rb:286:in `block in prepare_folders'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/host.rb:285:in `each'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/host.rb:285:in `prepare_folders'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/message/initialize.rb:14:in `block in process'
~/.asdf/installs/ruby/2.7.1/lib/ruby/2.7.0/benchmark.rb:293:in `measure'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/message/initialize.rb:10:in `process'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/host.rb:101:in `receive'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/transport/adapter.rb:44:in `process'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/transport/adapter.rb:17:in `block in opening'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/transport/data_reader.rb:57:in `parse_message_from_buffer'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/transport/data_reader.rb:35:in `block in receive'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/transport/data_reader.rb:30:in `each_char'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/transport/data_reader.rb:30:in `receive'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/solargraph-0.38.6/lib/solargraph/language_server/transport/adapter.rb:27:in `receiving'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/backport-1.1.2/lib/backport/client.rb:63:in `tick'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/backport-1.1.2/lib/backport/server/stdio.rb:18:in `update'
~/.asdf/installs/ruby/2.7.1/lib/ruby/2.7.0/observer.rb:197:in `block in notify_observers'
~/.asdf/installs/ruby/2.7.1/lib/ruby/2.7.0/observer.rb:196:in `each'
~/.asdf/installs/ruby/2.7.1/lib/ruby/2.7.0/observer.rb:196:in `notify_observers'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/backport-1.1.2/lib/backport/client.rb:121:in `read_input'
~/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/backport-1.1.2/lib/backport/client.rb:102:in `block in run_input_thread'
```

**Comment by castwide** [comment]
Fixed in the `v0.39` branch. It'll be released in v0.39.0.

**Comment by aaronjensen** [comment]
That's great, thanks! Will you close this issue then when it's released?

**Comment by castwide** [comment]
Sure, no problem.

**Comment by castwide** [comment]
Released in v0.39.0.
